docs/preview/features/secret-store/index.md

```diff
@@ -78,6 +77,34 @@ public class HealthController : ControllerBase
 }
 ```
 
+### Using secret store outside .NET hosting
+The secret store is also available directly on the `IServiceCollection` for applications that run outside the .NET hosting but still want to use the Arcus secret store.
+
+Just like you would register the secret store on the `HostBuilder`, you can use the `.AddSecretStore` extension to register the secret store:
+
+```csharp
+IServiceCollection services = ...
+IConfiguration configuration = ...
+
+// Mark: the secret store uses logging, so make sure that you have set the logging beforehand.
+service.AddLogging(...);
```

Comment:
Not sure what to do here. Should we make sure that the secret store can be set up without logging, or should we fail on missing logging infrastructure in the services?

Reply:
My feelings are that you should not impede requirements / dependencies to the consumer of the library.   In other words, although logging is mostly required in projects, usage of the library should not fail if the consumer has not setup logging.

Reply:
I've talked about this with @tomkerkhove  a few days ago regarding logging in libraries.  Logging can be very handy here when troubleshooting, but in normal circumstances, I believe that logs from a library should be as concise as possible.   Libraries should log by default to a Trace / Verbose level imho.

Reply:
Fully agree on both topics

Reply:
@stijnmoreels we should change our secret store so that if there is no logger we use a nulllogger

Reply:
This is maybe also something that should be opted-in/opted-out for in general, but we can discuss that in case that's a good idea

Reply:
Yes, I also agree on this. I've changed it so logging is only used when it's available.